traits/src/status_type.rs

```diff
@@ -0,0 +1,21 @@
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+#[derive(Encode, Decode, Debug, Clone, PartialEq, TypeInfo)]
+pub enum StatusType {
+    ClaimSpirits,
+    PurchaseRareEggs,
+    PreorderEggs,
+}
+
+#[allow(clippy::upper_case_acronyms)]
+impl StatusType {
+    pub fn from_u8(value: u8) -> Option<StatusType> {
```

Comment:
I believe we don't actually need `from_u8`. Most of the place we can just use the enum as is. Sometimes we may need to encode the enum as a byte blobl (`Vec<u8>` with a length of 1), and it can be done by `Encode::encode` and `Decode::decode`. We may not need to convert it to the plain `u8` in the lifetime.